### DISCUSSION TITLE
Composite Entities

### DISCUSSION THREAD

**Discussion by jcuenod** [explosion/spaCy#6829] [Help: Coding & Implementations] [answered]
I'm migrating from `wit.ai` because they've decided to force facebook signups.

I am tagging natural language searches like `find all the clauses containing plural verbs and singular nouns`. To do this, I've been making extensive use of [Composite Entities](https://medium.com/wit-ai/introducing-composite-entities-ba2639a26e0) in `wit.ai`. Tagging would be something like:

- `plural verbs` → `search_query`
- `plural` → `number`
- `verbs` → `part_of_speech`
- `singular nouns` → `search_query`
- `singular` → `number`
- `nouns` → `part_of_speech`

I see that spaCy does not appear to support composite entities (#1334). To my mind, this problem needs a nested solution because otherwise differentiating the queries would be impossible. Is there some other way to implement this? Or does spaCy have some alternative to composite entities?

**Comment by jcuenod** [comment, marked as answer]
["Overlapping spans"](https://twitter.com/_inesmontani/status/1352555571728109568) looks promising...

**Reply by svlandeg** [reply]
You can find the relevant work-in-progress here: https://github.com/explosion/spaCy/pull/6747 and https://github.com/explosion/spaCy/pull/6696

**Reply by jcuenod** [reply]
Ahh, so this is still a WIP, not actually part of v3?

**Reply by svlandeg** [reply]
The required datastructure is already released as part of v3.0, cf https://spacy.io/api/spangroup so you should definitely be able to use that already. In addition, Matt's been working on a `SpanCategorizer` component that would provide similar functionality to our NER component but then for potentially overlapping spans. The latter is not released yet.
